Emscripten Support

7coil opened the pull request:
Take a peek.

https://webspeak.terminal.ink/

7coil committed:
emscripten: Get Emscripten builds working

7coil committed:
emscripten: Create temporary user interface

7coil committed:
emscripten: Add Bootstrap to DECtalk for Web

7coil commented:
I'll be back in a week to consider my life choices.

stargo commented on src/dapi/src/cmd/cm_text.c:
```diff
@@ -792,27 +794,27 @@ void cm_text_getclause(LPTTS_HANDLE_T phTTS)
 										{
 										case 'a':
-											pCmd_t->clausebuf[j]=(unsigned char)'�';
+											pCmd_t->clausebuf[j]=(unsigned char)'�';
```
This will break german umlauts, they need to be iso8859-1/windows-1252 in the code.

stargo commented on src/dapi/src/cmd/cm_text.c:
```diff
@@ -147,11 +147,12 @@ extern const unsigned short parser_char_types[];
 extern int _far par_dict_lookup(PKSD_T, char *, int);
 #endif
 
+#ifndef NEW_BINARY_PARSER
```
This now breaks the build on linux. Need to find out what is really going on here...

stargo commented on src/dapi/src/cmd/cm_copt.c:
```diff
@@ -1913,9 +1913,9 @@ int cm_cmd_power(LPTTS_HANDLE_T phTTS)
 				   cm_util_say_string(pKsd_t, "Battery okay. ", 1);
 				if (pKsd_t->lang_curr == LANG_spanish)
-					cm_util_say_string(pKsd_t, "Bater�a corriendo bien. ", 1);
+					cm_util_say_string(pKsd_t, "Bater�a corriendo bien. ", 1);
```
It's quite unlikely this code gets built for the dectalk express again, but this and the following accented char changes would break battery status for spanish/latin american

stargo commented on src/dapi/src/cmd/par_def1.h:
```diff
@@ -1,9 +1,9 @@
- *                           Copyright �
- *	  Copyright � 2000-2001 Force Computers Inc., a Solectron company. All rights reserved.
- *    � Digital Equipment Corporation 1995, 1998. All rights reserved.
+ *                           Copyright �
```
File has no changes except whitespace and the broken copyright. Maybe not include it in this PR.

stargo commented on src/dapi/src/lts/ls_task.c:
```diff
@@ -1743,27 +1743,27 @@ int ls_task_fix_german_vowels(PLTS_T pLts_t, LETTER *llp, LETTER **rlp)
 			{
 			case 'a':
-				elp->l_ch=(unsigned char)'�';
+				elp->l_ch=(unsigned char)'?';
```
More breakage of german umlauts here and below.

stargo commented on src/dapi/src/lts/ls_task.c:
```diff
@@ -3877,7 +3877,7 @@ int ls_task_plain_number_processing(LPTTS_HANDLE_T phTTS, LETTER *llp, LETTER *r
 			}
 #else
-		if (lp2+3 == rlp && lp2->l_ch==(unsigned char)'�' && (lp2+1)->l_ch=='m' 
+		if (lp2+3 == rlp && lp2->l_ch==(unsigned char)'?' && (lp2+1)->l_ch=='m' 
```
Breakage of accented and other chars here and below

stargo commented on src/samples/SAY/say.c:
```diff
@@ -2,7 +2,7 @@
 *  COPYRIGHT NOTICE                                                     	
 *   
-*  Copyright � 2001 Force Computers, Inc., a Solectron Company. All rights reserved.	
+*  Copyright � 2001 Force Computers, Inc., a Solectron Company. All rights reserved.	
```
This only changes the copyright-char, maybe not include this file in this PR?

stargo commented on src/samplosf/src/dtsamples/say.c:
```diff
@@ -822,6 +833,13 @@ int main( int argc, char *argv[] )
     if ( TextToSpeechShutdown( ttsHandle ) != MMSYSERR_NOERROR )
 	fprintf(stderr,"TextToSpeechShutdown failed.\n");
 
+#ifdef __EMSCRIPTEN__
```
Should this be here?

Commit:
fix special characters

They need to be ISO-8859-1/Windows-1252 and can't be UTF-8.

stargo reviewed (commented):
Only two comments left now, maybe the parser-one needs to be resolved after merge.

Commit:
cm_text: rename par_* functions and calls to not cause conflicts

These are defined as inline in different files and as inline gets
redefined to nothing in emscripten builds this causes duplicate
definitions. Just rename the implementations in this file to avoid
this.
